api/src/utils/legacy/legacyTangleHelper.ts

```diff
@@ -41,17 +41,22 @@ export class LegacyTangleHelper {
         try {
             const client = new LegacyClient(network.provider, network.user, network.password);
 
+            let maxResults = limit || network.apiMaxResults;
+            if (maxResults > network.apiMaxResults) {
+                maxResults = network.apiMaxResults;
+            }
+
             const response = await client.findTransactions({
                 ...findReq,
-                maxresults: 5000,
+                maxresults: maxResults,
             });
 
             if (response?.txHashes && response.txHashes.length > 0) {
                 hashes = response.txHashes;
                 cursor.node = hashes.length;
 
                 if (limit === undefined) {
-                    cursor.hasMore = hashes.length === 5000;
+                    cursor.hasMore = hashes.length === maxResults;
```

Comment:
You sure about this logic. The "old" line looks weird with the specific number (5000). I guess the reasoning of it is that if it's a "full page" from the node, it means there's more (?).
Are you sure the new configurable thing fits into this logic ?

Reply:
the backend doesn't have pagination. we increased the limit in the backend now to 10000, I guess when there is more than that, it's useless to display more anyway :D